packages/node_modules/@webex/webex-core/src/interceptors/default-options.js

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) 2015-2019 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {Interceptor} from '@webex/http-core';
+
+
+/**
+ * Allows the user of the SDK to set default options that apply every http request made by the SDK
+ * For example a default timeout could be set for all requests like this :
+ *
+ *   webex = WebexSdk.init({
+ *     credentials: {
+ *       supertoken: superToken
+ *     },
+ *     config: {
+ *       credentials: {
+ *         client_id,
+ *         client_secret
+ *       },
+ *       defaultRequestOptions: {
+ *         timeout: 15000
+ *       }
+ *     }
+ *   });
+ */
+export default class DefaultOptionsInterceptor extends Interceptor {
+  /**
+   * @returns {DefaultOptionsInterceptor}
+   */
+  static create() {
+    return new DefaultOptionsInterceptor({webex: this});
+  }
+
+  /**
+   * @see Interceptor#onRequest
+   * @param {Object} options
+   * @returns {Object}
+   */
+  onRequest(options) {
+    const {defaultRequestOptions: defaultOptions} = this.webex.config;
+
+    if (!defaultOptions) {
+      return options;
+    }
+
+    Object.keys(defaultOptions).forEach((key) => {
+      // don't override any existing option properties
+      if (!Object.keys(options).includes(key)) {
+        options[key] = defaultOptions[key];
```

Comment:
Is there additional intent to append default header key:values in default options? If so, this may not work here, as it does not deep merge/check.

Reply:
the use cases i had was only for top level properties so didn't want to over complicate it, but if you think a deep merge would be better i can look into it.